kratos/python_scripts/eigen_solver_factory.py

```diff
@@ -1,31 +1,69 @@
+import KratosMultiphysics.kratos_utilities as kratos_utils
+
+if kratos_utils.CheckIfApplicationsAvailable("EigenSolversApplication"):
+    import KratosMultiphysics.EigenSolversApplication as EiSA
 
-    if(type(settings) != KratosMultiphysics.Parameters):
+if kratos_utils.CheckIfApplicationsAvailable("ExternalSolversApplication"):
+    import KratosMultiphysics.ExternalSolversApplication as ExSA
+
+def ConstructSolver(settings):
+    if not isinstance(settings, KM.Parameters):
         raise Exception("Input is expected to be provided as a Kratos Parameters object")
 
     solver_type = settings["solver_type"].GetString()
 
     if solver_type == "eigen_eigensystem":
-        import KratosMultiphysics.EigenSolversApplication
-        eigen_solver = KratosMultiphysics.EigenSolversApplication.EigensystemSolver(settings)
-        return eigen_solver
-
-    linear_solver = linear_solver_factory.ConstructSolver(settings["linear_solver_settings"])
-
-    if(solver_type == "power_iteration_eigenvalue_solver"):
-        eigen_solver = KratosMultiphysics.PowerIterationEigenvalueSolver( settings, linear_solver)
-    elif(solver_type == "power_iteration_highest_eigenvalue_solver"):
-        eigen_solver = KratosMultiphysics.PowerIterationHighestEigenvalueSolver( settings, linear_solver)
-    elif(solver_type == "rayleigh_quotient_iteration_eigenvalue_solver"):
-        eigen_solver = KratosMultiphysics.RayleighQuotientIterationEigenvalueSolver( settings, linear_solver)
-    elif(solver_type == "FEAST" or solver_type == "feast"):
-        import KratosMultiphysics.ExternalSolversApplication
-        eigen_solver = KratosMultiphysics.ExternalSolversApplication.FEASTSolver(settings, linear_solver)
+        if kratos_utils.CheckIfApplicationsAvailable("EigenSolversApplication"):
+            eigen_solver = EiSA.EigensystemSolver(settings)
+            return eigen_solver
+        else:
+            raise Exception("EigenSolversApplication not available")
+
+    linear_solver = CreateLinearSolver(settings)
+
+    if solver_type == "power_iteration_eigenvalue_solver":
+        eigen_solver = KM.PowerIterationEigenvalueSolver( settings, linear_solver)
+    elif solver_type == "power_iteration_highest_eigenvalue_solver":
+        eigen_solver = KM.PowerIterationHighestEigenvalueSolver( settings, linear_solver)
+    elif solver_type == "rayleigh_quotient_iteration_eigenvalue_solver":
+        eigen_solver = KM.RayleighQuotientIterationEigenvalueSolver( settings, linear_solver)
+    elif solver_type == "FEAST" or solver_type == "feast":
+        if kratos_utils.CheckIfApplicationsAvailable("ExternalSolversApplication"):
+            eigen_solver = ExSA.FEASTSolver(settings, linear_solver)
+        else:
+            raise Exception("ExternalSolversApplication not available")
     else:
         raise Exception("Solver type not found. Asking for :" + solver_type)
 
     return eigen_solver
+
+def CreateLinearSolver(settings):
```

Comment:
```suggestion
def CreateDirectLinearSolver(settings):
```

Reply:
We should move this to a common utility perhaps, it is basically copied from the mechanical_solver.py

Reply:
I also thought this
I suggest to move it to the python-lin-solver-factory

Reply:
I was thinking the same